_studio/shared/src/cm_mem_copy.cpp

```diff
@@ -3164,6 +3165,59 @@ mfxStatus CmCopyWrapper::CopyMirrorVideoToVideoMemory(void *pDst, void *pSrc, mf
 
 #define CM_RGB_SUPPORTED_COPY_SIZE(ROI) (ROI.width <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT && ROI.height <= CM_RGB_MAX_GPUCOPY_SURFACE_HEIGHT )
 
+bool CmCopyWrapper::CheckSurfaceContinuouslyAllocated(mfxFrameSurface1 *surf)
+{
+    mfxFrameData* ptrFrameData = &surf->Data;
+
+    switch (surf->Info.FourCC)
+    {
+    case MFX_FOURCC_P010:
+    case MFX_FOURCC_P016:
+    case MFX_FOURCC_P210:
+    case MFX_FOURCC_NV16:
+    case MFX_FOURCC_NV12:
+        {
+            mfxU32 stride_in_bytes = ptrFrameData->PitchLow + ((mfxU32)ptrFrameData->PitchHigh << 16);
+
+            size_t totalBufferUPSize = stride_in_bytes * ALIGN32(surf->Info.Height)
+                + stride_in_bytes * surf->Info.Height / 2;
+
+            if (ptrFrameData->UV + stride_in_bytes * surf->Info.Height / 2 > ptrFrameData->Y + totalBufferUPSize)
+            {
+                return false;
+            }
+
+            break;
+        }
+
+    case MFX_FOURCC_YV12:
```

Comment:
Absolutely agree with comments from @lakulako We already discovered most of them with @asmirno1 in offline discussion. I will just point one more time that for planar formats like YV12 we must check all three Y/U/V pointers